CustomerRegister/Repositories/CustomerService.cs

```diff
@@ -0,0 +1,80 @@
+using CustomerRegister.Repositories.Interfaces;
+using Microsoft.AspNetCore.Http.Features;
+using Microsoft.AspNetCore.Http.HttpResults;
+using Microsoft.IdentityModel.SecurityTokenService;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace CustomerRegister
+{
+    public class CustomerService : ICustomerService
+    {
+        private readonly List<CustomerEntity> _customers = new();
+
+        public bool Exists(CustomerEntity selectedCustomer)
+        {
+            var response = _customers.Any(customer => customer.Cpf == selectedCustomer.Cpf || customer.Email == selectedCustomer.Email) && (selectedCustomer.Id != selectedCustomer.Id);
+            return response;
+        }
+
+        public bool DuplicatedRegister(CustomerEntity selectedCustomer)
+        {
+            if(_customers.Any(customer => customer.Cpf == selectedCustomer.Cpf || customer.Email == selectedCustomer.Email))
+                {
+                return true;
+                }
+            return false;
+        }
+
+        public int AddCustomer(CustomerEntity customer)
+        {
+            customer.Id = _customers.LastOrDefault()?.Id + 1 ?? 1;
+
+            if (Exists(customer)) return 0;
+
+            if(DuplicatedRegister(customer)) return 0;
+
+            _customers.Add(customer);
+            return customer.Id;
+        }
+
+        public bool DeleteCustomer(int id)
```

Comment:
Você já tá na service de customer, acho que não faz sentido adicionar customer junto dos nomes dos métodos
```suggestion
        public bool Delete(int id)
```

Reply:
feito